docs/addons/opa-gatekeeper.md

```diff
@@ -0,0 +1,195 @@
+- input.request.operation specifies the type of the operation, i.e., CREATE, UPDATE, DELETE, CONNECT.
+- input.request.userInfo specifies the identity of the caller.
+- input.request.object contains the entire Kubernetes object.
+- input.request.oldObject specifies the previous version of the Kubernetes object on UPDATE and DELETE.
+
+The policies you give to OPA ultimately generate an admission review response that is sent back to the API Server. Gatekeeper 
+
+
+### What are Pod Security Policies (PSP's)?
+Pod Security Policies (PSP's) is a common use case you will see being defined by Gatekeeper within a Kubernetes cluster. A Pod Security Policy (PSP) is a cluster-level resource for managing security aspects of a pod specification. PSP's allow you to control things like the ability to run privileged containers or the ability to control access to host filesystems. The example below shows a policy that is assigned to all authenticated users in a cluster which limits volume types, denies running as root/escalating to root, requires a security profile, and a few other aspects.
+
+```yaml
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: restricted
+  annotations:
+    seccomp.security.alpha.kubernetes.io/allowedProfileNames: 'docker/default'
+    apparmor.security.beta.kubernetes.io/allowedProfileNames: 'runtime/default'
+    seccomp.security.alpha.kubernetes.io/defaultProfileName:  'docker/default'
+    apparmor.security.beta.kubernetes.io/defaultProfileName:  'runtime/default'
+spec:
+  privileged: false
+  # Required to prevent escalations to root.
+  allowPrivilegeEscalation: false
+  # This is redundant with non-root + disallow privilege escalation,
+  # but we can provide it for defense in depth.
+  requiredDropCapabilities:
+    - ALL
+  # Allow core volume types.
+  volumes:
+    - 'configMap'
+    - 'emptyDir'
+    - 'projected'
+    - 'secret'
+    - 'downwardAPI'
+    # Assume that persistentVolumes set up by the cluster admin are safe to use.
+    - 'persistentVolumeClaim'
+  hostNetwork: false
+  hostIPC: false
+  hostPID: false
+  runAsUser:
+    # Require the container to run without root privileges.
+    rule: 'MustRunAsNonRoot'
+  seLinux:
+    # This policy assumes the nodes are using AppArmor rather than SELinux.
+    rule: 'RunAsAny'
+  supplementalGroups:
+    rule: 'MustRunAs'
+    ranges:
+      # Forbid adding the root group.
+      - min: 1
+        max: 65535
+  fsGroup:
+    rule: 'MustRunAs'
+    ranges:
+      # Forbid adding the root group.
+      - min: 1
+        max: 65535
+  readOnlyRootFilesystem: false
+```
+
+
+**Note: PSP's will be deprecated as of Kubernetes version 1.21 so please keep that in mind while you are evaluating this add on. To learn more please follow this [link](https://kubernetes.io/blog/2021/04/06/podsecuritypolicy-deprecation-past-present-and-future/)**
+
+## Getting Started with OPA Gatekeeper
```

Comment:
We need to call out any limitations. E.g. it looks like it is Linux only, things like that.  Will it work on Fargate? Let's add a section and call those out.

Reply:
@parkand1 ^^

Reply:
addressed